xdsl/dialects/arm_neon.py

```diff
@@ -123,9 +130,95 @@ def assembly_str(self):
             return f"{self.reg.register_name.data}.{self.arrangement.data.name}[{self.index}]"
 
 
+@irdl_op_definition
+class GetRegisterOp(ARMOperation):
+    """
+    This instruction allows us to create an SSAValue for a given register name.
+    """
+
+    name = "arm_neon.get_register"
+
+    result = result_def(NEONRegisterType)
+    assembly_format = "attr-dict `:` type($result)"
+
+    def __init__(self, register_type: NEONRegisterType):
+        super().__init__(result_types=[register_type])
+
+    def assembly_line(self):
+        return None
+
+
+@irdl_op_definition
+class DSSFMulVecScalarOp(ARMInstruction):
+    """
+    Floating-point multiply (mixed: first source operand is a vector, second is a scalar. Destination is a vector)
+    This instruction multiplies each of the floating-point values in the first source operand by the
+    second source operand and writes the resulting values to the corresponding lanes of the destination.
+    Encoding: FMUL <Vd>.<T>, <Vn>.<T>, <Vm>.<idx>.
+    Vd, Vn, Vm specify the regs. The <T> specifier determines element arrangement (size and count).
+    The <idx> specifier determines the index of Vm at which the second source operand (scalar) can be found,
+    preceded by a size specifier.
+    https://developer.arm.com/documentation/ddi0602/2024-12/SIMD-FP-Instructions/FMUL--vector---Floating-point-multiply--vector--?lang=en#T_option__4
+    """
+
+    name = "arm_neon.dss.fmulvec"
+    d = result_def(NEONRegisterType)
+    s1 = operand_def(NEONRegisterType)
+    s2 = operand_def(NEONRegisterType)
+    scalar_idx = attr_def(IntegerAttr[i8])
+    arrangement = attr_def(NeonArrangementAttr)
+
+    assembly_format = (
+        "$s1 `,` $s2 attr-dict `:` `(` type($s1) `,` type($s2) `)` `->` type($d)"
+    )
+
+    def __init__(
+        self,
+        s1: Operation | SSAValue,
+        s2: Operation | SSAValue,
+        *,
+        d: NEONRegisterType,
+        arrangement: NeonArrangement | NeonArrangementAttr,
+        comment: str | StringAttr | None = None,
+    ):
+        if isinstance(comment, str):
+            comment = StringAttr(comment)
+        if isinstance(arrangement, NeonArrangement):
+            arrangement = NeonArrangementAttr(arrangement)
+        super().__init__(
+            operands=(s1, s2),
+            attributes={
+                "comment": comment,
+                "arrangement": arrangement,
+            },
+            result_types=(d,),
+        )
+
+    def assembly_instruction_name(self) -> str:
+        return "fmul"
+
+    def assembly_line_args(self):
+        assert isinstance(self.d.type, NEONRegisterType)
+        assert isinstance(self.s1.type, NEONRegisterType)
+        assert isinstance(self.s2.type, NEONRegisterType)
```

Comment:
I was hoping that the `VectorWithArrangement` class would do this for us, to avoid extra calls inside the helper

Reply:
it definitely doesn't work out-of-the-box, I've been trying a few things but haven't found a way to make it work yet.
```
Argument of type "Attribute" cannot be assigned to parameter "reg" of type "NEONRegisterType | SSAValue" in function "__init__"
  Type "Attribute" is not assignable to type "NEONRegisterType | SSAValue"
    "Attribute" is not assignable to "NEONRegisterType"
    "Attribute" is not assignable to "SSAValue"
    ```

Reply:
it seems like it's complaining about `self.d.type` instead of `self.d`

Reply:
if I change ```self.d.type``` to ```self.d``` in the argument to VectorWithArrangement, the assertion fails with: ```AssertionError: Expected NEONRegisterType, but got OpResult```